### PR TITLE
chore: Pages dropdown common components

### LINKED ISSUE #33048: [Task]: IDE - Common components for pages dropdown

### PR DESCRIPTION
**Pull request by ashit-rath**
## Description
Extracts certain building blocks for the IDE into common components to introduce more reusability in implementation of IDE in other editors

This PR deals with the pages dropdown section only. 

Fixes #33048

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8919287328>
> Commit: 4d534b4e8df1f240a8f164089893aca8673553e3
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8919287328&attempt=2" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: app/client/src/IDE/Structure/EditorHeaderDropdown.test.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        <EditorHeaderDropdown.Body>
+          <span>Body</span>
+        </EditorHeaderDropdown.Body>
+      </EditorHeaderDropdown>,
+    );
+
+    expect(getByText("Header")).toBeInTheDocument();
+    expect(getByText("Body")).toBeInTheDocument();
+  });
+});
```
[the remark this answers is not in the document]
Comment (ashit-rath) [reply app/client/src/IDE/Structure/EditorHeaderDropdown.test.tsx:22]: There are no props to be passed to the components present in the file.

File: app/client/src/IDE/index.ts [hunk 2]
```diff
@@ -10,6 +10,14 @@
  * It is possible to use the IDE Header without using these subsections
  */
 export { default as IDEHeader } from "./Structure/Header";
+// Better naming, figure with Hetu
```
Comment (hetunandu) [inline app/client/src/IDE/index.ts:13]: While we are using the dropdown in the Editor mode, it dont think we should make this restricted by the name. We can call this `HeaderDropdown` and eventually refactor the "OrgName/AppName" component to this as well. The current hold back is that the Trigger also needs to be an `EditableName` component.

Also I feel this is a UI component and should be merged in with the `IDEHeaderEditorSwitcher` component. What are your thoughts on this?

Comment (ashit-rath) [reply app/client/src/IDE/index.ts:13]: Understood and makes sense

**Review by albinAppsmith — APPROVED** [review]

**Pull request merged**